src/neuroconv/datainterfaces/behavior/audio/audiointerface.py

```diff
@@ -0,0 +1,203 @@
+                for file_path in self.source_data["file_paths"]
+            ]
+        )
+        metadata = super().get_metadata()
+        metadata["Behavior"] = behavior_metadata
+
+        return metadata
+
+    def add_acoustic_waveform_series(
+        self,
+        nwbfile: NWBFile,
+        metadata: dict,
+        file_path: str,
+        starting_time: float,
+        write_as: Optional[str] = "stimulus",  # "stimulus" or "acquisition"
+        stub_test: bool = False,
+        stub_frames: int = 1000,
+        iterator_options: Optional[dict] = None,
+        compression_options: Optional[dict] = None,
+    ):
+
+        compression_options = compression_options or dict(compression="gzip")
+        iterator_options = iterator_options or dict()
+
+        container = nwbfile.acquisition if write_as == "acquisition" else nwbfile.stimulus
+        # Early return if acoustic waveform series with this name already exists in NWBFile
+        if metadata["name"] in container:
+            return
+
+        # Load the audio file
+        sampling_rate, audio_data = read(filename=file_path, mmap=True)
+        if stub_test:
+            audio_data = audio_data[:stub_frames]
+
+        acoustic_waveform_series_kwargs = dict(
+            rate=float(sampling_rate),
+            starting_time=starting_time,
+            data=H5DataIO(SliceableDataChunkIterator(data=audio_data, **iterator_options), **compression_options)
+            if not stub_test
+            else audio_data,
+        )
+
+        # Add metadata
+        acoustic_waveform_series_kwargs.update(**metadata)
+
+        # Create AcousticWaveformSeries with ndx-sound
+        acoustic_waveform_series = AcousticWaveformSeries(**acoustic_waveform_series_kwargs)
+
+        # Add audio recording to nwbfile as acquisition or stimuli
+        if write_as == "acquisition":
+            nwbfile.add_acquisition(acoustic_waveform_series)
+        elif write_as == "stimulus":
+            nwbfile.add_stimulus(acoustic_waveform_series)
+
+        return nwbfile
+
+    def run_conversion(
+        self,
+        nwbfile_path: OptionalFilePathType = None,
+        nwbfile: Optional[NWBFile] = None,
+        metadata: Optional[dict] = None,
+        stub_test: bool = False,
+        write_as: str = "stimulus",
+        starting_times: Optional[list] = None,
+        iterator_options: Optional[dict] = None,
+        compression_options: Optional[dict] = None,
+        overwrite: bool = False,
+        verbose: bool = True,
+    ):
+
+        file_paths = self.source_data["file_paths"]
+
+        assert write_as in ["stimulus", "acquisition"], "Audio can be written either as 'stimulus' or 'acquisition'."
+
+        if metadata is None:
+            metadata = dict()
+
+        audio_metadata = metadata.get("Behavior", dict()).get("Audio", None)
+        if audio_metadata is None:
+            audio_metadata = self.get_metadata()["Behavior"]["Audio"]
```

Comment:
Correct me if I'm wrong, but I had thought our typical workflow for defaulting metadata like this was performed in the corresponding tools function, and the interfaces job was just to pass inputs to that? That way if someone just wants to use the tool without metadata, it will figure it out at that level